### DISCUSSION TITLE
How does tauri::command work? IPC?

### DISCUSSION THREAD

**Discussion by ospfranco** [tauri-apps/tauri#8957] [Q&A]
I'm starting to do some Tauri development and I was reading through the docs, specifically the [Calling Rust from the frontend](https://tauri.app/v1/guides/features/command) work. And I was just curious how does it actually work? Does it serialize the call from JS to JSON? Does it work via OS IPC? Thanks!

**Comment by FabianLars** [comment]
For v1 we once started to writing something here: <https://tauri.app/v1/references/architecture/inter-process-communication/#commands> it's pretty basic but the v1 ipc itself is pretty basic itself.

The OS is not involved, the webviews just offer "postMessage()" apis we use to pass the messages, but in v1 serializing everything to json strings is indeed required.

The new ipc in v2 is a mix between the old one and [custom protocols](https://docs.rs/tauri/latest/tauri/struct.Builder.html#method.register_uri_scheme_protocol) (basically http/network requests with fetch that don't actually hit the network). This is faster and more flexible (for example it allow binary payloads)

**Reply by ospfranco** [reply]
Thanks for the reply!
